config/v1/types_authentication.go

```diff
@@ -195,74 +195,118 @@ const (
 )
 
 type OIDCProvider struct {
-	// name of the OIDC provider
+	// name configures the unique human-readable identifier
+	// associated with the identity provider.
+	// name is used to distinguish between multiple identity providers.
+	// name has no impact on token validation or authentication mechanics.
+	// name is required.
+	// name must not be an empty string ("").
 	//
 	// +kubebuilder:validation:MinLength=1
 	// +required
 	Name string `json:"name"`
-	// issuer describes atributes of the OIDC token issuer
+
+	// issuer configures how the platform interacts
+	// with the identity provider and how tokens issued
+	// from the identity provider are evaluated by the
+	// Kubernetes API Server.
+	// issuer is required.
 	//
 	// +required
 	Issuer TokenIssuer `json:"issuer"`
 
-	// oidcClients contains configuration for the platform's clients that
-	// need to request tokens from the issuer
+	// oidcClients configures how on-cluster, platform clients should request tokens from the identity provider.
+	// oidcClients is optional.
+	// oidcClients must not exceed 20 entries.
+	// oidcClients entries must have unique namespace/name pairs.
 	//
 	// +listType=map
 	// +listMapKey=componentNamespace
 	// +listMapKey=componentName
 	// +kubebuilder:validation:MaxItems=20
+	// +optional
 	OIDCClients []OIDCClientConfig `json:"oidcClients"`
 
-	// claimMappings describes rules on how to transform information from an
-	// ID token into a cluster identity
+	// claimMappings configures the rules to be used by the Kubernetes API Server
+	// for translating claims in a JWT token, issued by the identity provider,
+	// to a cluster identity.
+	// claimMappings is optional.
+	// +optional
 	ClaimMappings TokenClaimMappings `json:"claimMappings"`
 
-	// claimValidationRules are rules that are applied to validate token claims to authenticate users.
+	// claimValidationRules configures the rules to be used the Kubernetes API Server
+	// for validating the claims in a JWT token issued by the identity provider.
+	// claimValidationRules is optional.
+	//
+	// Validation rules are joined via an AND operation.
 	//
 	// +listType=atomic
+	// +optional
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
 }
 
 // +kubebuilder:validation:MinLength=1
 type TokenAudience string
 
 type TokenIssuer struct {
-	// URL is the serving URL of the token issuer.
-	// Must use the https:// scheme.
+	// issuerURL configures the URL that is used to issue tokens
```

Comment:
Wrong field name.

Reply:
`issuerURL` is the serialized name which we prefer to use for GoDoc on the OpenShift APIs because it is what end-users will be familiar with. The serialized field name is what the documentation, generated by the GoDoc, will be attached to when users do something like `oc explain ...` on the field